canonical/cloud-init
# fix(net): Make duplicate route add succeed.

## Proposed Commit Message
```
fix(net): Make duplicate route add succeed.

This behaves the same but doesn't fail when adding an existing route.

Fixes GH-3441
Fixes GH-3595
```

## Additional Context
Fixes #3441 
Fixes #3595 

## Test Steps
Once can manually verify that `ip route replace` behaves as documented.

old:
```bash
sudo ip route add 192.168.1.0/24 dev enp24s0 proto kernel scope link src 192.168.0.104 metric 100
sudo ip route add 192.168.1.0/24 dev enp24s0 proto kernel scope link src 192.168.0.104 metric 100  # fails
```

new:
```bash
sudo ip route replace 192.168.1.0/24 dev enp24s0 proto kernel scope link src 192.168.0.104 metric 100
sudo ip route replace 192.168.1.0/24 dev enp24s0 proto kernel scope link src 192.168.0.104 metric 100  # succeeds
```
